File: PHP/formatTimeInSecs.php
<?php
/*
 * @author Amir Sanni <amirsanni@gmail.com>
 * @description Function takes a time in sec and return it in a more human friendly format (e.g. 10hrs, 20mins)
 * @date 11th March, 2016
 * @param $timeInSec the time to format (in secs)
 */
function formatTimeInSecs($timeInSec){
  $hour = (int)($timeInSec/3600);//get the number of hours
        
  $hourString = $hour >= 1 ? $hour."h" : "";//attach 'h' to hour if hour is not less than one, else set string as empty.
        
  $secsLeft = $timeInSec%3600;//get the number of secs left after removing the hours which is equivalent to the modulus of $timeInSec
  
  if($secsLeft < 60){
    $minuteString = "0m";//if the number of secs is less than 60, we set minute to 0 meaning the number of secs left is not up to a min
    $secString = $secsLeft ."s";//and set secs to the number of secs
    
    $minAndSecString = $minuteString . " " . $secString;//then concatenate the two strings
  }
  
  else{//else, we get the number of minutes left and the number of seconds left
    $minute = (int) ($secsLeft / 60); //get the number of minutes left
    $minuteString = $minute >= 1 ? $minute . "m" : "";

    $sec = $secsLeft % 60; //get the number of secs left
    $secString = $sec ? $sec . "s" : "";

    $minAndSecString = $minuteString . " " . $secString;
  }
  
  //return the formatted time
  return $hourString . " " . $minAndSecString;
}

File: python/formatTimeInSecs.py
def formatTimeInSec(timeInSec):
  if timeInSec >= 3600: #time is one hour and above
    time = hrMinSec(timeInSec)

  elif timeInSec >= 60:#time is a minute and above but not upto one hour
    time = minuteAndSec(timeInSec)

  else: #it is not upto a minute
    time = str(timeInSec) + "s"

  return time


#function will be called only if time is a minute or more but not upto an hr
def minuteAndSec(timeInSec):
  minute = timeInSec/60 #get the number of minutes

  minstr = str(minute) + "m"

  sec = timeInSec % 60 #get the number of secs (if any)

  if sec:
    secstr = str(sec) + "s"

  else:
    secstr = ""

  time = minstr+" "+secstr

  return time


#function will be called only if time is an hour or more
def hrMinSec(timeInSec):
  hr = timeInSec/3600

  if hr:
    hrstr = str(hr) + "h"

  else:
    hrstr = ""

  secsleft = timeInSec % 3600

  if secsleft < 60:
    minstr = "" #set min string as empty since the ses left is not up to a minute

    if secsleft > 0: #only add seconds if there is at least one sec left
      secstr = str(secsleft) + "s"
    else:
      secstr = ""

    minandsecstr = minstr + " " + secstr #concatenate the two strings

  else: #if the number of secs left is upto a minute or more
    minandsecstr = minuteAndSec(secsleft)

  return hrstr+" "+minandsecstr



while True:
  print("Input time (in seconds) or letter q to quit")
  inputtedTime = input()

  if inputtedTime == "q":
    break

  else:
    print(formatTimeInSec(inputtedTime))
